lib/subscribers.go

```diff
@@ -77,4 +82,85 @@ func (s *SubscriberService) Delete(ctx context.Context, subscriberID string) (Su
 	return resp, nil
 }
 
+func (s *SubscriberService) GetNotificationFeed(ctx context.Context, subscriberID string, opts *SubscriberNotificationFeedOptions) (*SubscriberNotificationFeedResponse, error) {
+	var resp SubscriberNotificationFeedResponse
+	URL := s.client.config.BackendURL.JoinPath("subscribers", subscriberID, "notifications", "feed")
+
+	if opts != nil {
+		queryValues := URL.Query()
+		if opts.Page != nil {
+			queryValues.Add("page", fmt.Sprint(*opts.Page))
+		}
+
+		if opts.FeedIdentifier != nil {
+			queryValues.Add("feedIdentifier", *opts.FeedIdentifier)
+		}
+
+		if opts.Seen != nil {
+			queryValues.Add("seen", strconv.FormatBool(*opts.Seen))
+		}
+
+		URL.RawQuery = queryValues.Encode()
+	}
```

Comment:
Is it possible to make it dynamic so that additional keys can be added in the future without requiring any updates to the code?

Reply:
The first thing that comes to mind is to make the query list a map and iterate over it, but that sacrifices strong typing of struct. Will check if we can leverage some other go feature for this.

Reply:
Yeah, this works. Although, with this approach, the caller of the library function has to know beforehand which key and values can be passed to the function (by going to the API docs). Basically, we are offloading the work of constructing the input struct to the caller.

Reply:
@prajjwaldimri please can you make the update? Thanks 🙏🏿

Reply:
@samsoft00 Figured out a way to have the best of both worlds. Check out https://github.com/argonautdev/go-novu/blob/99e0e0d10f832eaa2666b736bf5908b5eb871fe1/lib/utils.go#L24

Also, please review.